frontend/src/Core/ElmStreet.elm

```diff
@@ -42,3 +45,8 @@ elmStreetDecodePair decA decB = D.map2 Tuple.pair (D.index 0 decA) (D.index 1 de
 elmStreetDecodeTriple : Decoder a -> Decoder b -> Decoder c -> Decoder (a, b, c)
 elmStreetDecodeTriple decA decB decC = D.map3 (\a b c -> (a,b,c)) (D.index 0 decA) (D.index 1 decB) (D.index 2 decC)
 
+elmStreetDecodeNonEmpty : Decoder a -> Decoder (a, List a)
+elmStreetDecodeNonEmpty decA = D.list decA |> D.andThen (\xs -> case xs of
+                                                 h::t -> D.succeed (h, t)
+                                                 _    -> D.fail "Expecting at least ONE ELEMENT array")
```

Comment:
we probably should improve formatting of this.

Reply:
Feel free to elm-format this (probably in separate PR)